google-cloud-datastore/src/test/java/com/google/cloud/datastore/AggregationQueryTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.datastore;
+
+import static com.google.cloud.datastore.AggregationQuery.Mode.GQL;
+import static com.google.cloud.datastore.AggregationQuery.Mode.STRUCTURED;
+import static com.google.cloud.datastore.StructuredQuery.PropertyFilter.eq;
+import static com.google.cloud.datastore.aggregation.Aggregation.count;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThrows;
+
+import com.google.cloud.datastore.aggregation.CountAggregation;
+import com.google.common.collect.ImmutableSet;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.function.ThrowingRunnable;
+import org.junit.rules.ExpectedException;
+
+public class AggregationQueryTest {
+
+  private static final String KIND = "Task";
+  private static final String NAMESPACE = "ns";
+  private static final EntityQuery COMPLETED_TASK_QUERY = Query.newEntityQueryBuilder()
+      .setNamespace(NAMESPACE)
+      .setKind(KIND)
+      .setFilter(eq("done", true))
+      .build();
+
+  @Rule
+  public ExpectedException exceptionRule = ExpectedException.none();
+
+  @Test
+  public void testAggregations() {
+    AggregationQuery aggregationQuery = Query.newAggregationQueryBuilder()
+        .setNamespace(NAMESPACE)
+        .addAggregation(new CountAggregation("total_upto_100", 100))
```

Comment:
The [design doc](http://go/java-datastore-count-design#heading=h.nxddek4w5dli) mentions this:

```
  .addAggregation(count().as("total"))           // not CountAggregation
```

Reply:
Yes, both syntax are supported.

https://github.com/googleapis/java-datastore/pull/823#discussion_r971588252

The next test in the same file follows the builder construct.